tests/streaming/test_streaming.py

```diff
@@ -134,6 +134,100 @@ def module_start_status_message(self, instance, inputs):
         assert status_messages[2].message == "Predict starting!"
 
 
+@pytest.mark.anyio
+async def test_default_then_custom_status_message_provider():
```

Comment:
We may want to refactor the testing a bit - instead of default => custom, we can put up two threads that apply different `MyStatusMessageProvider`, and verify that both threads work as expected. Then at the end we verify that the callback is not being modified.